### PR TITLE
met analytics api

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Description of changes:*
New API for analytics database

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: analytics-api/src/analytics_api/config.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    # JWT_OIDC Settings
+    JWT_OIDC_WELL_KNOWN_CONFIG = os.getenv('JWT_OIDC_WELL_KNOWN_CONFIG')
+    JWT_OIDC_ALGORITHMS = os.getenv('JWT_OIDC_ALGORITHMS', 'RS256')
+    JWT_OIDC_JWKS_URI = os.getenv('JWT_OIDC_JWKS_URI')
+    JWT_OIDC_ISSUER = os.getenv('JWT_OIDC_ISSUER')
+    JWT_OIDC_AUDIENCE = os.getenv('JWT_OIDC_AUDIENCE', 'account')
+    JWT_OIDC_CACHING_ENABLED = os.getenv('JWT_OIDC_CACHING_ENABLED', 'True')
+    JWT_OIDC_JWKS_CACHE_TIMEOUT = 300
+
+    S3_BUCKET = os.getenv('S3_BUCKET')
```
Comment (tiagograf-AOT) [inline analytics-api/src/analytics_api/config.py:81]: lets remove all these environment variables not being used (only keep the ones you have in the sample env)

File: analytics-api/src/analytics_api/resources/__init__.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+That are used to expose operational health information about the service, and meta information.
+"""
+
+from flask import Blueprint
+
+from .apihelper import Api
+from .engagement import API as ENGAGEMENT_API
+__all__ = ('API_BLUEPRINT',)
+
+URL_PREFIX = '/api/'
```
Comment (tiagograf-AOT) [inline analytics-api/src/analytics_api/resources/__init__.py:30]: i think we are dropping the url prefix, i'm not sure yet

File: analytics-api/setup.cfg [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+keywords =
+
+[options]
+zip_safe = True
+python_requires = >=3.6
+include_package_data = True
+packages = find:
+
+[options.package_data]
+met_api =
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply analytics-api/setup.cfg]: changed it, thanks.

File: analytics-api/src/analytics_api/__init__.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+from flask import Flask, g, request
+from flask_cors import CORS
+
+from analytics_api.auth import jwt
+from analytics_api.config import get_named_config
+from analytics_api.models import db, ma, migrate
+
+
+# Security Response headers
+csp = (
```
[the remark this answers is not in the document]
Comment (VineetBala-AOT) [reply analytics-api/src/analytics_api/__init__.py]: removed

File: analytics-api/migrations/versions/05e82fada6fe_add_runcycle_id_to_all_tables.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = '05e82fada6fe'
+down_revision = 'e62ba34dc486'
+branch_labels = None
```
Comment (saravanpa-aot) [inline analytics-api/migrations/versions/05e82fada6fe_add_runcycle_id_to_all_tables.py:15]: i hope these revision numbers are correct and the scripts wont run when we deploy

**Review by saravanpa-aot — APPROVED** [review]

File: analytics-api/src/analytics_api/resources/__init__.py (hunk 2 above)
Comment (tiagograf-AOT) [reply analytics-api/src/analytics_api/resources/__init__.py:30]: lets keep it

Comment (VineetBala-AOT) [reply analytics-api/src/analytics_api/resources/__init__.py:30]: ok

File: analytics-api/migrations/versions/05e82fada6fe_add_runcycle_id_to_all_tables.py (hunk 5 above)
Comment (VineetBala-AOT) [reply analytics-api/migrations/versions/05e82fada6fe_add_runcycle_id_to_all_tables.py:15]: Yes its the same we had within met-cron.

**Review by tiagograf-AOT — APPROVED** [review]

**Pull request merged**